src/main/java/redis/clients/jedis/util/CaffeineCSC.java

```diff
@@ -0,0 +1,95 @@
+package redis.clients.jedis.util;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import java.util.concurrent.TimeUnit;
+import net.openhft.hashing.LongHashFunction;
+import redis.clients.jedis.ClientSideCache;
+import redis.clients.jedis.CommandObject;
+import redis.clients.jedis.args.Rawable;
+
+public class CaffeineCSC extends ClientSideCache {
+
+  private static final int DEFAULT_MAXIMUM_SIZE = 10_000;
+  private static final int DEFAULT_EXPIRE_SECONDS = 100;
+  private static final LongHashFunction DEFAULT_HASH_FUNCTION = LongHashFunction.xx3();
```

Comment:
IMHO let's share with users a comment on why this function. Be kind to our future selves and community.

Reply:
Sure... we'll also need why Caffeine, why Guava, why not only one, etc.

Reply:
100%